### PR TITLE
Grade detail 115

### LINKED ISSUE #115: Detailed grade
When an instructor gives a grade based on several sub-criteria, the student only gets the overall grade (sum of sub-criteria) as feed-back. He should be able to see the details.

### PR DESCRIPTION
**Pull request by brendon-lind**
Added Grade Detail to student homework overview.

Resolves #115 

Test with Jupiter enabled/disabled Test with multiple criteria. 

# Checklist
- [x] Code review by me 
- [x] Hand tested by me 
- [x] I'm proud of my work
- [x] Code review by reviewer
- [x] Hand tested by reviewer
- [x] Ready to merge

### CODE REVIEW

**Comment by brendon-lind** [comment]
Screen Shot 2020-04-20 at 4 04 07 PM

File: apps/homework/templates/homework/overview.html [hunk 1]
```diff
@@ -15,6 +15,22 @@
         .no-padding {
             padding: 1vh 0 0 0 !important;
         }
+        div.ui.statistic {
+            padding: 1em 0 0 0;
+            width: 100%;
+        }
+
+        div.ui.tiny.statistic {
+            width: 100%;
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html]: Do we need width defined twice here?

File: apps/homework/templates/homework/overview.html [hunk 2]
```diff
@@ -213,11 +233,39 @@ <h4 class="ui header">Submission Details:</h4>
...
+                                                        {{ grade.text_grade }}
+                                                      </div>
+                                                      <div class="label">
+                                                        Total Grade
+                                                      </div>
+                                                    </div>
+                                                        {% if definition.jupyter_notebook_enabled %}
+                                                            <div class="ui tiny statistic">
+                                                                <div class="value">
+                                                                    {{  grade.jupyter_notebook_grade }}/{{ definition.jupyter_notebook_highest }}
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html]: Remove extra space character

File: apps/homework/templates/homework/overview.html [hunk 3]
```diff
@@ -213,11 +233,39 @@ <h4 class="ui header">Submission Details:</h4>
...
+                                                                </div>
+                                                            </div>
+                                                        {% endif %}
+                                                        {% for criteria_answer in grade.criteria_answers.all %}
+                                                            <div class="ui tiny statistic">
+                                                                <div class="value">
+                                                                    {{ criteria_answer.score }}/{{ criteria_answer.criteria.upper_range }}
+                                                                </div>
+                                                                <div class="label">
+                                                                    {{criteria_answer.criteria.description}}
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html]: add space character between symbol and brackets for stylistic consistency

File: apps/homework/templates/homework/overview.html [hunk 4]
```diff
@@ -213,11 +233,39 @@ <h4 class="ui header">Submission Details:</h4>
...
-                                                        <h4 class="ui header">Teacher comments: </h4>
+                                                <div class="five wide column">
+                                                    <div class="ui statistic">
+                                                      <div class="value">
+                                                        {{ grade.text_grade }}
+                                                      </div>
+                                                      <div class="label">
+                                                        Total Grade
+                                                      </div>
+                                                    </div>
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html]: These look like they have 2-space indent while the rest has 4-space indent. Change to 4-space.

File: apps/homework/templates/homework/overview.html [hunk 5]
```diff
@@ -213,11 +233,39 @@ <h4 class="ui header">Submission Details:</h4>
...
+                                                        {% for criteria_answer in grade.criteria_answers.all %}
+                                                            <div class="ui tiny statistic">
+                                                                <div class="value">
+                                                                    {{ criteria_answer.score }}/{{ criteria_answer.criteria.upper_range }}
+                                                                </div>
+                                                                <div class="label">
+                                                                    {{criteria_answer.criteria.description}}
+                                                                </div>
+                                                            </div>
+                                                        {% endfor %}
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html]: Looks like these are indented too far

File: apps/homework/models.py [hunk 6]
```diff
@@ -206,7 +206,7 @@ class Grade(models.Model):
     published = models.BooleanField(default=False)
 
     needs_review = models.BooleanField(default=True)
-    jupyter_notebook_grade = models.FloatField(null=True)
```
Comment (gibsonbailey) [inline apps/homework/models.py:209]: This should be nullable

File: apps/homework/templates/homework/overview.html [hunk 7]
```diff
@@ -15,6 +15,22 @@
         .no-padding {
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html]: These css classes, `.no-padding` and `.condensed`, are defined but not used in multiple files including this one! Please address this.

**Review by gibsonbailey — CHANGES_REQUESTED** [review]
I'll take another look at this when these comments have been addressed. Thanks @brendon-lind!

File: apps/homework/templates/homework/overview.html (hunk 1 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html]: OK gotcha.

File: apps/homework/templates/homework/overview.html (hunk 2 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html]: Removed.

File: apps/homework/templates/homework/overview.html (hunk 3 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html]: Spaced.

File: apps/homework/templates/homework/overview.html (hunk 4 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html]: Done.

File: apps/homework/templates/homework/overview.html (hunk 5 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html]: Not anymore!

File: apps/homework/models.py (hunk 6 above)
Comment (brendon-lind) [reply apps/homework/models.py:209]: Nulled

File: apps/homework/templates/homework/overview.html (hunk 7 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html]: Removed from offending files. Placed in actual file where needed.

File: apps/homework/templates/homework/overview.html [hunk 8]
```diff
@@ -7,14 +7,24 @@
 
 {% block extra_head %}
     <style>
-        .condensed {
-            margin: 0.5vh 0 0 0 !important;
-            padding: 0.5vh 0 0 0 !important;
+        div.ui.statistic {
+            padding: 1em 0 0 0;
+            width: 100%;
         }
```
Comment (gibsonbailey) [inline apps/homework/templates/homework/overview.html:14]: See here, there's one line separating this css class from the one following it. We should have uniform formatting for all of these css classes, and I'd like to see a new line between each of them.

File: apps/klasses/templates/klasses/klass_header.html [hunk 9]
```diff
@@ -1,5 +1,16 @@
 {% load static %}
 
+<style>
+    .condensed {
+            margin: 0.5vh 0 0 0 !important;
+            padding: 0.5vh 0 0 0 !important;
+        }
+
```
Comment (gibsonbailey) [inline apps/klasses/templates/klasses/klass_header.html:8]: Thanks for consolidating these styles into a single location! This should improve the maintainability of the project 😁

**Review by gibsonbailey — CHANGES_REQUESTED** [review]
Just a couple more spacing nit-picks left for you, then it'll be mergeable!

File: apps/homework/templates/homework/overview.html (hunk 8 above)
Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html:14]: Gotcha. Fixed now.

Comment (brendon-lind) [reply apps/homework/templates/homework/overview.html:14]: Gotcha. Fixed now

File: apps/klasses/templates/klasses/klass_header.html (hunk 9 above)
Comment (brendon-lind) [reply apps/klasses/templates/klasses/klass_header.html:8]: Welcome!

**Pull request merged**